### PR TITLE
OLS-538: Unit test for too long prompt exception

### PR DESCRIPTION
**Pull request by tisnik**
## Description

[OLS-538](https://issues.redhat.com//browse/OLS-538): Unit test for too long prompt exception

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [x] Unit test

## Related Tickets & Documents

- Related Issue #[OLS-538](https://issues.redhat.com//browse/OLS-538)
- Closes #

### CODE REVIEW

File: tests/unit/app/endpoints/test_ols.py [hunk 1]
```diff
@@ -173,6 +174,20 @@ def test_validate_question_valid_kw(llm_validate_question_mock, _load_config):
...
 
 
+@patch(
+    "ols.src.query_helpers.question_validator.QuestionValidator.validate_question",
+    side_effect=PromptTooLongError("Prompt length 10000 exceeds LLM"),
+)
+def test_validate_question_too_long_query(llm_validate_question_mock, _load_config):
+    """Check the behaviour of validate_question function with too long query."""
+    conversation_id = suid.get_suid()
+    query = "Tell me about Kubernetes?" * 10000
```
Comment (bparees) [inline tests/unit/app/endpoints/test_ols.py:184]: is this necessary?  you've already setup a side effect that validate_question is going to raise the exception regardless.

File: tests/unit/app/endpoints/test_ols.py [hunk 2]
```diff
@@ -173,6 +174,20 @@ def test_validate_question_valid_kw(llm_validate_question_mock, _load_config):
...
+    llm_request = LLMRequest(query=query, conversation_id=conversation_id)
+    # PromptTooLongError should be caught and HTTPException needs to be raised
+    with pytest.raises(HTTPException, match="413: {'response': 'Prompt is too long'"):
+        ols.validate_question(conversation_id, llm_request)
```
Comment (bparees) [inline tests/unit/app/endpoints/test_ols.py:188]: what is this really testing?  you setup a mock such that the exception is guaranteed to be raised.  So at best this is just testing that ols.validate_question is catching and reraising that exception?

**Review by bparees — COMMENTED** [review]
this test is useful, but not as critical to me as a test that validates that we detect and handle a too long prompt (by actually submitting a prompt that is too long, not by mocking raising an exception).

Also there are two cases to cover:

1) initial question that goes through question validation
2) follow up question because it won't go through question validation

In order to ensure that prompts that are too long get handled in both cases (one case where question validation is processing the prompt, the other in the case where docs summarizer is)

those would be e2e tests of course.

File: tests/unit/app/endpoints/test_ols.py (hunk 2 above)
Comment (tisnik) [reply tests/unit/app/endpoints/test_ols.py:188]: just this, it's UT, nothing on integration nor e2e level now

File: tests/unit/app/endpoints/test_ols.py (hunk 1 above)
Comment (tisnik) [reply tests/unit/app/endpoints/test_ols.py:184]: not right now

Comment (bparees) [reply tests/unit/app/endpoints/test_ols.py]: please remove it then, because as it is now it's misleading as to what the test is actually doing/testing.

Comment (tisnik) [reply tests/unit/app/endpoints/test_ols.py]: just did it :)

File: tests/unit/app/endpoints/test_ols.py (hunk 2 above)
Comment (tisnik) [reply tests/unit/app/endpoints/test_ols.py:188]: for record, e2e waits for https://github.com/openshift/lightspeed-service/pull/820 and then https://github.com/openshift/lightspeed-service/pull/810 . when these two will be in, e2e should be created with real end-to-end behavior checking

**Comment by bparees** [comment]
/approve
/lgtm

but i remain skeptical of the value of tests like this, or rather the value relative to the effort to write+maintain them.  Effectively this test just confirms that ols.validate_question catches an exception that is raised by question_validator.validate_question.  (and it's worth noting that question_validator doesn't even normally raise the exception itself, it's raised by a helper that question_validator calls.  So if the behavior of question_validator were changed to not call that helper, this test would continue to pass despite not serving any purpose.  That's somewhat a general limitation of unit tests, but somewhat a particular limitation of this specific test and its use of mocks)

**Pull request merged**